zulip/integrations/twitter/twitter-bot

```diff
@@ -259,14 +258,47 @@ for status in statuses[::-1][:opts.limit_tweets]:
     elif opts.twitter_name:
         subject = composed
 
-    message = {
-        "type": "stream",
-        "to": [opts.stream],
-        "subject": subject,
-        "content": url
-    }
-
-    ret = client.send_message(message)
+    # If the tweet is a retweet then check if the original tweet has been sent before
+    # and try to edit it, else post new message with the original tweet.
+    # This contains all strings that could have caused the tweet to match our query.
+    message_id = 0
+    edit_message = False
+    if status.retweeted_status is not None:
+        try:
+            message_id = config_internal.getint('twitter', str(status.retweeted_status.id))
+            edit_message = True
+        except (NoOptionError, NoSectionError):
+            edit_message = False
+        url = (status.retweeted_status.urls[0].expanded_url + "\n") + ("Retweets: %s" % (status.retweeted_status.retweet_count,))
+    else:
+        url = url + "\nRetweets: " + str(status.retweet_count)
+    # If we are editing message it implies the current tweet is a retweet
+    # and has been posted before. Hence try to edit the older tweet and
+    # change its number of retweets but if it has passed its edit time
+    # do nothing.
+    if edit_message is True:
+        message_data = {
+            "message_id": message_id,
+            "content": url,
+        }
+        ret = client.update_message(message_data)
+
+    if edit_message is False:
+        message = {
+            "type": "stream",
+            "to": [opts.stream],
+            "subject": subject,
+            "content": url
+        }
+
+        ret = client.send_message(message)
+        if ret['result'] == 'success':
+            if 'twitter' not in config_internal.sections():
+                config_internal.add_section('twitter')
+            if status.retweeted_status is None:
+                config_internal.set('twitter', str(status.id), str(ret['id']))
+            else:
+                config_internal.set('twitter', str(status.retweeted_status.id), str(ret['id']))
```

Comment:
This means that the config file content will grow with the number of retweeted messages?

Reply:
Yes. I am using it as a kind of database.

Reply:
Maybe the message id should be stored in a separate dedicated key-value database?

Reply:
Yes that would be better. Can you suggest what I should use for the key-value database?

Reply:
Can't come up with any solution on top of my head. Haven't seen any being used by python-zulip-api bots before. But if other bots will become more complex to require a db, they might be using the same db we would use here.

Reply:
After a quick search, I found pickledb, a nosql key-value db in pure python. Unfortunately I can't be decisive about this, given that there are several alternatives. Maybe we should discuss in czo.